lnclient/lnd/lnd.go

```diff
@@ -492,6 +503,60 @@ func (svc *LNDService) MakeInvoice(ctx context.Context, amount int64, descriptio
 	return transaction, nil
 }
 
+func (svc *LNDService) getRoutingHints(ctx context.Context) (hints []*lnrpc.RouteHint, err error) {
+	channelsRes, err := svc.client.ListChannels(ctx, &lnrpc.ListChannelsRequest{
+		PrivateOnly: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	processedPeers := make(map[string]struct{})
+	for _, channel := range channelsRes.Channels {
+		if _, ok := processedPeers[channel.RemotePubkey]; ok {
+			continue
+		}
+
+		chanInfo, err := svc.client.GetChanInfo(ctx, &lnrpc.ChanInfoRequest{
+			ChanId: channel.ChanId,
+		})
+		if err != nil {
+			logger.Logger.WithFields(logrus.Fields{
+				"channel_id": channel.ChanId,
+			}).WithError(err).Error("Unable to add routing hint")
+			continue
+		}
+
+		remotePolicy := chanInfo.Node2Policy
```

Comment:
Node2 is not always the counterparty is it?

Reply:
Ah true, will add a check using node pubkey